cli/cppcheckexecutor.cpp

```diff
@@ -186,30 +186,40 @@ bool CppCheckExecutor::parseFromArgs(Settings &settings, int argc, const char* c
         logger.printMessage("Please use --suppress for ignoring results from the header files.");
     }
 
-    const std::vector<std::string>& pathnames = parser.getPathNames();
-    const std::list<FileSettings>& fileSettings = parser.getFileSettings();
+    const std::vector<std::string>& pathnamesRef = parser.getPathNames();
+    const std::list<FileSettings>& fileSettingsRef = parser.getFileSettings();
 
     // the inputs can only be used exclusively - CmdLineParser should already handle this
-    assert(!(!pathnames.empty() && !fileSettings.empty()));
+    assert(!(!pathnamesRef.empty() && !fileSettingsRef.empty()));
 
-    if (!fileSettings.empty()) {
+    if (!fileSettingsRef.empty()) {
+        std::list<FileSettings> fileSettings;
         if (!settings.fileFilters.empty()) {
             // filter only for the selected filenames from all project files
-            std::copy_if(fileSettings.cbegin(), fileSettings.cend(), std::back_inserter(mFileSettings), [&](const FileSettings &fs) {
+            std::copy_if(fileSettingsRef.cbegin(), fileSettingsRef.cend(), std::back_inserter(fileSettings), [&](const FileSettings &fs) {
                 return matchglobs(settings.fileFilters, fs.filename);
             });
-            if (mFileSettings.empty()) {
+            if (fileSettings.empty()) {
                 logger.printError("could not find any files matching the filter.");
                 return false;
             }
         }
         else {
-            mFileSettings = fileSettings;
+            fileSettings = fileSettingsRef;
         }
+
+        // sort the markup last
+        std::copy_if(fileSettings.cbegin(), fileSettings.cend(), std::back_inserter(mFileSettings), [&](const FileSettings &fs) {
```

Comment:
The approach with two loops is the same as the previous code. That should not be much of an issue since it is only done once in the main process. If anybody has an idea on how to do this with a single loop I would be happy to use that. But that doesn't have to be done in this PR,